### PR TITLE
LUKS suspend/resume and kill slot functions

### PR DESCRIPTION
**Pull request by vojtechtrefny**

### CODE REVIEW

File: src/lib/plugin_apis/crypto.api [hunk 1]
```diff
@@ -294,6 +295,43 @@ gboolean bd_crypto_luks_change_key_blob (const gchar *device, const guint8 *pass
...
 gboolean bd_crypto_luks_resize (const gchar *luks_device, guint64 size, GError **error);
 
+/**
+ * bd_crypto_luks_suspend:
+ * @luks_device: LUKS device to suspend
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether the given @luks_device was successfully suspended or not
+ *
+ * Tech category: %BD_CRYPTO_TECH_LUKS-%BD_CRYPTO_TECH_MODE_OPEN_CLOSE
```
Comment (vpodzime) [inline src/lib/plugin_apis/crypto.api]: Should use the new mode, right?

File: src/plugins/crypto.c [hunk 2]
```diff
@@ -962,6 +962,143 @@ gboolean bd_crypto_luks_resize (const gchar *luks_device, guint64 size, GError *
...
+            bd_utils_report_finished (progress_id, (*error)->message);
+            return FALSE;
+        }
+    } else
+        buf_len = data_len;
+
+    ret = crypt_resume_by_passphrase (cd, luks_device, CRYPT_ANY_SLOT,
+                                      key_buffer ? key_buffer : (char*) pass_data,
+                                      buf_len);
+    g_free (key_buffer);
```
Comment (vpodzime) [inline src/plugins/crypto.c]: This should probably also ``memset()`` the buffer to zeroes so that the key data isn't left in the memory.

**Comment by vojtechtrefny** [comment]
Jenkins, test this please.

**Comment by vojtechtrefny** [comment]
Jenkins, test this please.

**Comment by vojtechtrefny** [comment]
Jenkins, test this please.

**Pull request merged**